projects/client/RabbitMQ.Client/src/client/impl/Frame.cs

```diff
@@ -1,218 +1,312 @@
-                {
-                    throw ioe.InnerException;
-                }
-
-                throw;
-#else
-                // If it's a WSAETIMEDOUT SocketException, unwrap it.
-                // This might happen when the limit of half-open connections is
-                // reached.
-                if (ioe.InnerException == null ||
-                    !(ioe.InnerException is SocketException) ||
-                    ((SocketException)ioe.InnerException).SocketErrorCode != SocketError.TimedOut)
-                {
-                    throw ioe;
-                }
-                throw ioe.InnerException;
-#endif
-            }
-
-            if (type == 'A')
-            {
-                // Probably an AMQP protocol header, otherwise meaningless
-                ProcessProtocolHeader(reader);
-            }
-
-            int channel = reader.ReadUInt16();
-            int payloadSize = reader.ReadInt32(); // FIXME - throw exn on unreasonable value
-            byte[] payload = reader.ReadBytes(payloadSize);
-            if (payload.Length != payloadSize)
-            {
-                // Early EOF.
-                throw new MalformedFrameException("Short frame - expected " +
-                                                  payloadSize + " bytes, got " +
-                                                  payload.Length + " bytes");
-            }
-
-            int frameEndMarker = reader.ReadByte();
-            if (frameEndMarker != Constants.FrameEnd)
-            {
-                throw new MalformedFrameException("Bad frame end marker: " + frameEndMarker);
-            }
-
-            return new Frame(type, channel, payload);
-        }
-
-        public void FinishWriting()
-        {
-            if (m_accumulator != null)
-            {
-                Payload = m_accumulator.ToArray();
-                m_accumulator = null;
-            }
-        }
-
-        public NetworkBinaryReader GetReader()
-        {
-            return new NetworkBinaryReader(new MemoryStream(Payload));
-        }
-
-        public NetworkBinaryWriter GetWriter()
-        {
-            return new NetworkBinaryWriter(m_accumulator);
-        }
-
-        public override string ToString()
-        {
-            return base.ToString() + string.Format("(type={0}, channel={1}, {2} bytes of payload)",
-                Type,
-                Channel,
-                Payload == null
-                    ? "(null)"
-                    : Payload.Length.ToString());
-        }
-
-        public void WriteTo(NetworkBinaryWriter writer)
-        {
-            FinishWriting();
-            writer.Write((byte)Type);
-            writer.Write((ushort)Channel);
-            writer.Write((uint)Payload.Length);
-            writer.Write(Payload);
-            writer.Write((byte)Constants.FrameEnd);
-        }
-    }
-}
+// This source code is dual-licensed under the Apache License, version

```

Comment:
Why is the diff viewer showing a difference here?

Reply:
The delta between old and new has too many changes to allow the compare tool to provide a true comparison,  the frame class has been extended with many frame super classes.

Also the base frame class data is now immutable.

Reply:
It's actually easy to see what the difference is:

```
git log -p -- projects/client/RabbitMQ.Client/src/client/impl/Frame.cs
```

Existing line endings are replaced by Windows ones.